File: GA-2D/Genetic-Algorithm/libxl-4.2.0/examples/fortran/vs2019/extract/extract.f90
PROGRAM extract

USE libxl

TYPE(C_PTR) book, sheet, null
character(:), allocatable :: str
REAL(C_DOUBLE) d
INTEGER err
 
book = xlCreateXMLBook()
IF(xlBookLoad(book, "..\generate\example.xlsx").ne.0) THEN    
    sheet = xlBookGetSheet(book, 0)        
    str = xlSheetReadStr(sheet, 2, 1, null)
    PRINT *, str
    d = xlSheetReadNum(sheet, 3, 1, null)
    PRINT *, d        
END IF       
CALL xlBookRelease(book)

END PROGRAM

File: GA-2D/Genetic-Algorithm/libxl-4.2.0/examples/c/vs2019/extract/extract.c
#include <stdio.h>
#include "libxl.h"

int main()
{
    BookHandle book = xlCreateXMLBook();
    if(book) 
    {
        if(xlBookLoad(book, L"example.xlsx")) 
        {
            SheetHandle sheet = xlBookGetSheet(book, 0);
            if(sheet)
            {
                double d;
                const wchar_t* s = xlSheetReadStr(sheet, 2, 1, 0);

                if(s) wprintf(L"%s\n", s);

                d = xlSheetReadNum(sheet, 3, 1, 0);
                printf("%g\n", d);
            }
        }     
       
        xlBookRelease(book);
    }

    return 0;
}
